File: GerenciaMusic360.Entities/WorkCollaborator.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace GerenciaMusic360.Entities
{
    public partial class WorkCollaborator
    {
        public int Id { get; set; }
        public int WorkId { get; set; }
        public int ComposerId { get; set; }
        public decimal? AmountRevenue { get; set; }
        public decimal PercentageRevenue { get; set; }
        public bool? IsCollaborator { get; set; }
        public short? AssociationId { get; set; }
        [NotMapped]
        public Person Composer { get; set; }
        [NotMapped]
        public Work Work { get; set; }
        [NotMapped]
        public ComposerDetail ComposerDetail { get; set; }
        public Association Association { get; set; }
    }
}

File: GerenciaMusic360/ClientApp/app/shared/models/work-collaborator.ts
import { IPerson } from "./person";
import { Work } from "./work";
import { IComposerDetail } from "./composer-detail";
import { IAssociation } from "./association";

export interface IWorkCollaborator {
    id: number;
    workId: number;
    composerId: number;
    associationId: number;
    amountRevenue: number;
    percentageRevenue: number;
    composer?: IPerson;
    composerDetail?: IComposerDetail;
    association: IAssociation;
    work?: Work;
    isCollaborator: boolean;
}